iaas-common/src/main/java/io/github/cloudiator/persistance/PricingDomainRepository.java

```diff
@@ -0,0 +1,191 @@
+                .saveOrGet(domain.getOperatingSystem());
+
+        PricingModel pricingModel = new PricingModel(domain.id(), domain.providerId(), domain.name(),
+                domain.getInstanceName(), domain.getCloudServiceProviderName(), domain.getLicenseModel(),
+                domain.getLocationProviderId(), domain.getCurrency(), domain.getTenancy(), operatingSystemModel,
+                null,
+                apiModel, domain.getProductFamily(), domain.getPreInstalledSw(), domain.getCapacityStatus(), domain.getOperation());
+
+        List<PricingTermsModel> pricingOnDemandTermsModels = domain.getPricingOnDemandTerms().stream()
+                .map(term -> new PricingTermsModel(term.getLeaseContractLength(), term.getOfferingClass(),
+                        term.getPurchaseOption(),
+                        term.getPricingDimensions().stream()
+                                .map(dim -> new PricingPriceDimensionsModel(dim.getPricePerUnit(), dim.getUnit(), dim.getDescription(), dim.getBeginRange(), dim.getEndRange()))
+                                .collect(Collectors.toList()),
+                        "OnDemand", pricingModel))
+                .collect(Collectors.toList());
+
+        pricingOnDemandTermsModels.forEach(pricingTermsModel ->
+                pricingTermsModel.getPricingPriceDimensionsModels().forEach(pricingPriceDimensionsModel ->
+                        pricingPriceDimensionsModel.setPricingTermsModel(pricingTermsModel)));
+
+        List<PricingTermsModel> pricingReservedTermsModels = domain.getPricingReservedTerms().stream()
+                .map(term -> new PricingTermsModel(term.getLeaseContractLength(), term.getOfferingClass(),
+                        term.getPurchaseOption(),
+                        term.getPricingDimensions().stream()
+                                .map(dim -> new PricingPriceDimensionsModel(dim.getPricePerUnit(), dim.getUnit(), dim.getDescription(), dim.getBeginRange(), dim.getEndRange()))
+                                .collect(Collectors.toList()),
+                        "Reserved", pricingModel))
+                .collect(Collectors.toList());
+
+        pricingReservedTermsModels.forEach(pricingTermsModel ->
+                pricingTermsModel.getPricingPriceDimensionsModels().forEach(pricingPriceDimensionsModel ->
+                        pricingPriceDimensionsModel.setPricingTermsModel(pricingTermsModel)));
+
+        pricingModel.setPricingTermsModels(Stream.concat(pricingOnDemandTermsModels.stream(), pricingReservedTermsModels.stream()).collect(Collectors.toList()));
+
+        return pricingModel;
+    }
+
+    private ApiModel getApiModel(DiscoveredPricing domain) {
+        List<ApiModel> apiModels = apiDomainRepository.findAll();
+        Optional<ApiModel> apiModel = apiModels.stream().filter(model -> model.getProviderName().matches("(?i).*aws.*")).findAny();
+        if(apiModel.isPresent()) {
```

Comment:
Could be replaced by:
`apiModel.orElseGet(() -> apiDomainRepository.saveOrGet(domain.getApi()));`

Reply:
done